File: Problems/No.11_Container_With_Most_Water/container_with_most_water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        # store answer and the pointers
        CWMW_ans = 0
        local_ans = None
        left = 0
        right = len(height) - 1
        
        # start compute
        while (left < right):
            
            local_ans = right - left
            
            if (height[left] < height[right]):
                
                local_ans *= height[left]
                left += 1
            elif (height[left] > height[right]):
                local_ans *= height[right]
                right -= 1
            else:
                local_ans *= height[left]
                left += 1
                right -= 1
            
            if (local_ans > CWMW_ans):
                CWMW_ans = local_ans
            
        return CWMW_ans

File: Problems/No.11_Container_With_Most_Water/container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        // store the answer and the pointers
        int CWMW_ans = 0;
        int local_ans;
        int left = 0;
        int right = height.size() - 1;
        
        
        // start compute
        while (left < right) {
            
            local_ans = right - left;
            
            // case by case
            if (height[left] < height[right]) {
                
                local_ans *= height[left];
                left++;
                
            }
            else if (height[left] > height[right]) {
                
                local_ans *= height[right];
                right--;
                
            }
            else {
                
                local_ans *= height[left];
                left++;
                right--;
                
            }
            
            CWMW_ans = (local_ans>CWMW_ans) ? local_ans : CWMW_ans;
            
        }
        
        return CWMW_ans;
        
    }
};
